rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,712 @@
+          description: OK - Table Exists
+        401:
+          description: Unauthorized
+        404:
+          description: Not Found
+
+  /v1/tables/rename:
+    post:
+      tags:
+        - Catalog API
+      summary: Rename a table from its current name to a new name
+      description:
+        Rename a table from one identifier to another. It's valid to move a table
+        across namespaces, but the server implementation doesn't need to support it.
+      operationId: renameTable
+      requestBody:
+        description: Current table identifier to rename and new table identifier to rename to
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RenameTableRequest'
+        required: true
+      responses:
+        200:
+          description: OK
+        401:
+          description: Unauthorized
+        404:
+          description:
+            Not Found
+            - NoSuchTableException, Table to rename does not exist
+            - NoSuchNamespaceException, The target namespace of the new table identifier does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                TableToRenameDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+                NamespaceToRenameToDoesNotExist:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        406:
+          description: Not Acceptable - Unsupported Operation
+        409:
+          description: Conflict - The target table identifier to rename to already exists
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              example:
+                $ref: '#/components/examples/TableAlreadyExistsError'
+
+components:
+  #######################################################
+  # Common Parameter Definitions Used In Several Routes #
+  #######################################################
+  parameters:
+    namespace:
+      name: namespace
+      in: path
+      required: true
+      description:
+        A namespace identifier as a single string.
+        Multipart namespace parts should be separated by the null byte, %00.
+      schema:
+        type: string
+      examples:
+        singlepart_namespace:
+          value: "accounting"
+        multipart_namespace:
+          value: "accounting%00tax"
+
+    table:
+      name: table
+      in: path
+      description: A table name
+      required: true
+      schema:
+        type: string
+      example: "sales"
+
+    parent:
+      name: parent
+      in: query
+      description:
+        Optional parent namespace under which to list namespaces.
+        If parent is a multipart namespace, the parts must be separated by the null byte, %00.
+        When empty, list top-level namespaces.
```

Comment:
This looks like documentation for a use, not for the generic parameter.

Reply:
Yeah. I'm going to move this parameter back up, since it's only used in the one location and is not actually generic.

Reply:
This isn't a generic parameter. There's only one usage of it, and it's rather specific. So I've moved it to the one place it is used. If we wind up using it more than once (and thus introducing duplication), I'll move it back down here and then make it generic. But for the moment I think it's better as not a generic parameter. Feel free to suggest a generic description to use if you feel otherwise.